### PR TITLE
Update/sub control

### PR DESCRIPTION
**Pull request by Alan-mag**
The purpose of this pull request is to add a user 'Toolkit' to the submission site.

image

The purpose of the Toolkit is to provide a quick-access panel that allows volunteers/organizers to select the members of congress they are responsible for tracking and either autofill their information into the form, or quickly submit 'No Events.'

For each member of congress, the user can also save and edit specific URLs that they use to check on that particular member's Town Hall activity.

For Feedback:
@meganrm It would be great to get feedback on store/state structure as well as code organization

### CODE REVIEW

File: src/containers/UserToolkit/index.js [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+      uid,
+      userDisplayName,
+    };
+    currentTownHall.meetingType = 'No Events';
+    submitMetaData(metaData);
+    success();
+    return this.resetAll();
+  }
+
+  showModal = (type, link) => {
```
Comment (meganrm) [inline src/containers/UserToolkit/index.js]: This is hard to follow, why are these functions defined outside of a class? This also seems like it could be broken up into a few components.

File: src/state/user/actions.js [hunk 2]
```diff
@@ -7,10 +7,107 @@ const setUser = payload => ({
...
+  })
+}
+
+/******************* MOC LINK API******************************* */
+export const setSelectedMember = payload  => ({
+  payload,
+  type: 'SET_SELECTED_MEMBER',
+});
+
+export const getSelectedMemberInfo = payload => (dispatch) => {
```
Comment (meganrm) [inline src/state/user/actions.js]: this should be in the member-candidate branch of state

File: src/components/PageHeader/index.js [hunk 3]
```diff
@@ -15,13 +22,57 @@ const openNotification = () => {
...
-export default PageHeader;
\ No newline at end of file
+const mapStateToProps = state => ({
+  currentTownHall: getTownHall(state),
+});
+
+const mapDispatchToProps = dispatch => ({});
+
+PageHeader.propTypes = {
+  currentTownHall: PropTypes.shape({}).isRequired,
```
[the remark this answers is not in the document]
Comment (Alan-mag) [reply src/components/PageHeader/index.js]: Yeah, parent container 'App' isn't connected to state, so I will make PageHeader its own container.

[the remark this answers is not in the document]
Comment (meganrm) [reply src/components/PageHeader/index.js]: probably better to just make app connected to state. it's confusing if there are too many containers, probably already too many on this project

File: src/containers/UserToolkit/index.js (hunk 1 above)
Comment (Alan-mag) [reply src/containers/UserToolkit/index.js]: I have components for LinksList, Member Buttons, Edit form and Create form. Would you recommend a component for displaying messages?

File: src/state/user/actions.js (hunk 2 above)
Comment (Alan-mag) [reply src/state/user/actions.js]: both the member selection and links actions, right?

File: src/components/PageHeader/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Alan-mag) [reply src/components/PageHeader/index.js]: ok

File: src/state/user/actions.js (hunk 2 above)
Comment (meganrm) [reply src/state/user/actions.js]: No, the selected member should be in 'selections'. But if you're manipulating the moc data, then that should be in member-candidate.

Comment (Alan-mag) [reply src/state/user/actions.js]: Ok, I've separated a number of actions from users/ branch to member-candidate/ branch. The selected member and links is in member-candidate for now - the MOC link actions come from those selected states (add, edit, delete - selectedMoc and selectedLink)

File: src/containers/UserToolkit/index.js (hunk 1 above)
Comment (Alan-mag) [reply src/containers/UserToolkit/index.js]: I have added messages to class

Comment (meganrm) [reply src/containers/UserToolkit/index.js]: I mistakenly though these were outside the class since they were arrow functions. Please change them all to methods on the class for consistency 
Ie 
` showModal (type, link) {`

File: src/containers/UserToolkit/index.js [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+          handleAutoFillMember={this.handleAutoFillMember}
+        />
+      );
+    }
+  }
+
+  render() {
+    return (
+      <div>
+        <Card className="user-member-card" title={<span id="submitted-meta-data"><span>You've submitted: </span><span id="submitted-total">0</span><span> event(s)</span></span>}>
```
Comment (meganrm) [inline src/containers/UserToolkit/index.js]: Look at my most recent changes. This should now be coming from state.

Comment (Alan-mag) [reply src/containers/UserToolkit/index.js]: Ok, i've update eventCount received from state, thx!

File: src/state/user/actions.js [hunk 5]
```diff
@@ -29,8 +29,14 @@ export const writeUserData = payload => (dispatch) => {
     .then(dispatch(setUser(payload)));
 };
 
+// TODO: FIX - if no mocs
```
Comment (meganrm) [inline src/state/user/actions.js]: remove this todo
